pkg/storegateway/series_refs.go

```diff
@@ -829,15 +829,21 @@ func newLoadingSeriesChunkRefsSetIterator(
 }
 
 func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
-	sp, ctx := tracing.StartSpan(s.ctx, "loadingSeriesChunkRefsSetIterator.Next")
-	defer sp.Finish()
-
 	if s.err != nil {
 		return false
 	}
 	if !s.postingsSetIterator.Next() {
 		return false
 	}
+
+	// Start the span after the underlying iterator has been advanced, so that the span
+	// doesn't include the timing of the underlying iterator.
+	sp, ctx := tracing.StartSpan(s.ctx, "load_series_and_chunk_refs")
```

Comment:
I'm actually having doubts whether these (this + load_chunks) should be a span in the first place or just a log line which includes the duration of the operations.

With spans on requests with a lot of series we end up with massive parent spans in the store-gateway; basically multiple spans per 5000 series per block. 

I'm more inclined to the only log lines because they visualize better in tempo, but including the duration in them seemed a bit counter-intuitive. If reviewers agree I will make these simply log lines.

Reply:
> I'm more inclined to the only log lines because they visualize better in tempo, but including the duration in them seemed a bit counter-intuitive. If reviewers agree I will make these simply log lines.

That makes sense to me :+1:

Reply:
> If reviewers agree I will make these simply log lines.

Agree.

Reply:
👍 changed in https://github.com/grafana/mimir/pull/5294/commits/f38b63b883522aa1c2ed3f0c6748ab44aafd38c3

please take another look